swo_aws_extension/airtable/models.py

```diff
@@ -139,22 +139,36 @@ def get_available_mpa_from_pool(pls_enabled=False):
 
 def get_pending_notifications():
     """
-    Returns the pending notifications from the pool.
+    Returns the pending notifications from the pool for the selected PLS status.
+
     Returns:
-        list: The pending notifications.
+        list[PoolNotification]: The pending notifications for the selected PLS status.
     """
     pool_notification = get_pool_notification_model(AirTableBaseInfo.for_mpa_pool())
-    return pool_notification.all(EQUAL("Status", NotificationStatusEnum.PENDING))
+    pending_notifications = pool_notification.all(
+        formula=EQUAL(FIELD("Status"), STR_VALUE(NotificationStatusEnum.PENDING))
+    )
+    return pending_notifications
 
 
-def get_in_progress_notifications():
+def has_pending_notifications(pls_enabled):
     """
-    Returns the in progress notifications from the pool.
+    Returns whether there are pending notifications for the selected PLS status.
+
+    Args:
+        pls_enabled (bool): Whether the MPA has PLS enabled.
+
     Returns:
-        list: The in progress notifications.
+        bool: Whether there are pending notifications for the selected PLS status.
     """
     pool_notification = get_pool_notification_model(AirTableBaseInfo.for_mpa_pool())
-    return pool_notification.all(EQUAL("Status", NotificationStatusEnum.IN_PROGRESS))
+    pending_notifications = pool_notification.all(
```

Comment:
I would use here `first()` method. Otherwise you have to upload everything to memory first. Here if at least one exists that's true